source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -0,0 +1,345 @@
+
+        if (GlobalSettings.useAdvancedCardBottomTruncation) {
+            // Forces the container to be at least as tall as its content.
+            //
+            // Fixes a quirk in Chrome where, for nested flex elements, the
+            // inner element's height would never exceed the outer element's
+            // height. This caused overflow truncation to break -- containers
+            // would always be measured as not overflowing, since their heights
+            // were constrained by their parents as opposed to truly reflecting
+            // the height of their content.
+            //
+            // See the "Browser Rendering Notes" section of this answer:
+            // https://stackoverflow.com/questions/36247140/why-doesnt-flex-item-shrink-past-content-size
+            swiperWrapper.style.minHeight = '-webkit-min-content';
+        }
+
+        // switch (this.getEffectiveVerticalContentAlignment()) {
+        //     case VerticalAlignment.Center:
+        //         swiperWrapper.style.justifyContent = "center";
+        //         break;
+        //     case VerticalAlignment.Bottom:
+        //         swiperWrapper.style.justifyContent = "flex-end";
+        //         break;
+        //     default:
+        //         swiperWrapper.style.justifyContent = "flex-start";
+        //         break;
+        // }
+
+        const prevElementDiv: HTMLElement = document.createElement("button");
+        prevElementDiv.classList.add("swiper-button-prev");
+        swiperContainer.appendChild(prevElementDiv);
+
+        const nextElementDiv: HTMLElement = document.createElement("button");
+        nextElementDiv.classList.add("swiper-button-next");
+        swiperContainer.appendChild(nextElementDiv);
+
+        const pagination: HTMLElement = document.createElement("div");
+        pagination.classList.add(this.hostConfig.makeCssClassName("swiper-pagination"));
+        swiperContainer.appendChild(pagination);
+
+        const requestedNumberOfPages: number = Math.min(this._pages.length, this.hostConfig.carousel.maxCarouselPages);
+        if (this._pages.length > this.hostConfig.carousel.maxCarouselPages) {
+            console.warn(Strings.errors.tooManyCarouselPages);
+        }
+
+        if (this._pages.length > 0) {
+            for (let i = 0; i < requestedNumberOfPages; i++) {
+                const page = this._pages[i];
+                const renderedItem = this.isElementAllowed(page) ? page.render() : undefined;
+
+                if (renderedItem) {
+                    Utils.appendChild(swiperWrapper, renderedItem);
+                    this._renderedPages.push(page);
+                }
+            }
+        }
+
+        swiperContainer.appendChild(swiperWrapper as HTMLElement);
+
+        cardLevelContainer.appendChild(swiperContainer);
+
+        this.initializeSwiper(swiperContainer, nextElementDiv, prevElementDiv, pagination);
+
+        cardLevelContainer.onfocus = () => {
```

Comment:
@almedina-ms - instead of `onfocus`/`onmouseenter`/etc., we should be preferring [`addEventListener`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener). can you file an issue to track?

Reply:
https://github.com/microsoft/AdaptiveCards/issues/6701

Reply:
Issue opened

https://github.com/microsoft/AdaptiveCards/issues/6700